File: server/router/webcms/webconfig.go
package webcms

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type WebconfigRouter struct {
}

func (w *WebconfigRouter) InitWebConfigRouter(Router *gin.RouterGroup) {
	webconfigRouter := Router.Use(middleware.OperationRecord())
	var webconfigApi = v1.ApiGroupApp.WebcmsApiGroup.WebconfigApi
	{
		webconfigRouter.POST("/createWebconfig", webconfigApi.CreateWebconfig)
		webconfigRouter.GET("/getWebconfig", webconfigApi.GetWebconfig)
		webconfigRouter.DELETE("/delWebconfig", webconfigApi.DeleteWebconfig)
		webconfigRouter.POST("/setWebconfig", webconfigApi.SetWebconfig)
		webconfigRouter.POST("/changeWebconfig", webconfigApi.ChangeWebconfig)
	}
}

File: web/src/api/webconfig.js
import service from '@/utils/request'

export const getWebconfig = () => {
    return service({
      url: '/getWebconfig',
      method: 'get'
    })
}

export const setWebconfig = (data) => {
    return service({
      url: '/setWebconfig',
      method: 'post',
      data: data
    })
}

export const createWebconfig = (data) => {
  return service({
    url: '/createWebconfig',
    method: 'post',
    data
  })
}

export const delWebconfig = (data) => {
  return service({
    url: '/delWebconfig',
    method: 'delete',
    data
  })
}

// 切换站点
export const changeWebconfig = (data) => {
  return service({
    url: '/changeWebconfig',
    method: 'post',
    data: data
  })
}


